web-common/src/runtime-client/invalidation.ts

```diff
@@ -140,8 +149,19 @@ export function invalidateProfilingQueries(
   // do not re-fetch for failed entities.
   if (failed) return Promise.resolve();
 
-  return queryClient.refetchQueries({
-    predicate: (query) => isProfilingQuery(query.queryHash, name),
+  await queryClient.invalidateQueries({
+    predicate: (query) => {
+      // TODO: move the query matching to a separate module and reuse in http queue
+      const queryExtractorMatch = ProfilingQueryExtractor.exec(query.queryHash);
+      if (!queryExtractorMatch) return false;
+      const [, type, table] = queryExtractorMatch;
+      return table === name && type === "columns-profile";
+    },
+    type: "active",
+  });
```

Comment:
Similar to `isProfilingQuery`, this is hard to follow. Could one function contain the logic, and you could call it here?